test/compile_test.sh

```diff
@@ -41,47 +43,37 @@ capture_profile_d_export_script() {
 }
 
 test_simple() {
-  (
```

Comment:
What executes these tests? What's it looking for to determine a pass/failure?

Reply:
Heroku Buildpack Test runner (from `script/test` in this repo) https://github.com/heroku/heroku-buildpack-testrunner/blob/master/lib/test_utils.sh

In a subshell, the failures generated from `assert...` would not change state in the parent process so the result message always thought it had passed. The assertion failures were visible though.